modules/internal/ChapelBase.chpl

```diff
@@ -59,363 +59,363 @@ module ChapelBase {
   // a.type in the where clause is computed after instantiation.
   pragma "last resort"
-  inline proc =(ref a: borrowed class,   b: a.type) where b.type <= a.type { __primitive("=", a, b); }
+  inline operator =(ref a: borrowed class,   b: a.type) where b.type <= a.type { __primitive("=", a, b); }
   pragma "last resort"
-  inline proc =(ref a: borrowed class?,  b: a.type) where b.type <= a.type { __primitive("=", a, b); }
+  inline operator =(ref a: borrowed class?,  b: a.type) where b.type <= a.type { __primitive("=", a, b); }
   pragma "last resort"
-  inline proc =(ref a: unmanaged class,  b: a.type) where b.type <= a.type { __primitive("=", a, b); }
+  inline operator =(ref a: unmanaged class,  b: a.type) where b.type <= a.type { __primitive("=", a, b); }
   pragma "last resort"
-  inline proc =(ref a: unmanaged class?, b: a.type) where b.type <= a.type { __primitive("=", a, b); }
+  inline operator =(ref a: unmanaged class?, b: a.type) where b.type <= a.type { __primitive("=", a, b); }
 
-  inline proc =(ref a: nothing, b: ?t) where t != nothing {
+  inline operator =(ref a: nothing, b: ?t) where t != nothing {
     compilerError("a nothing variable cannot be assigned");
   }
 
-  inline proc =(ref a: ?t, b: nothing) where t != nothing {
+  inline operator =(ref a: ?t, b: nothing) where t != nothing {
     compilerError("cannot assign none to a variable of non-nothing type");
   }
 
   // This needs to be param so calls to it are removed after they are resolved
-  inline proc =(ref a: nothing, b: nothing) param { }
+  inline operator =(ref a: nothing, b: nothing) param { }
 
   //
   // equality comparison on primitive types
   //
-  inline proc ==(a: _nilType, b: _nilType) param return true;
-  inline proc ==(a: bool, b: bool) return __primitive("==", a, b);
-  inline proc ==(a: int(?w), b: int(w)) return __primitive("==", a, b);
-  inline proc ==(a: uint(?w), b: uint(w)) return __primitive("==", a, b);
-  inline proc ==(a: real(?w), b: real(w)) return __primitive("==", a, b);
-  inline proc ==(a: imag(?w), b: imag(w)) return __primitive("==", a, b);
-  inline proc ==(a: complex(?w), b: complex(w)) return a.re == b.re && a.im == b.im;
-  inline proc ==(a: borrowed object?, b: borrowed object?) return __primitive("ptr_eq", a, b);
-  inline proc ==(a: enum, b: enum) where (a.type == b.type) {
+  inline operator ==(a: _nilType, b: _nilType) param return true;
+  inline operator ==(a: bool, b: bool) return __primitive("==", a, b);
+  inline operator ==(a: int(?w), b: int(w)) return __primitive("==", a, b);
+  inline operator ==(a: uint(?w), b: uint(w)) return __primitive("==", a, b);
+  inline operator ==(a: real(?w), b: real(w)) return __primitive("==", a, b);
+  inline operator ==(a: imag(?w), b: imag(w)) return __primitive("==", a, b);
+  inline operator ==(a: complex(?w), b: complex(w)) return a.re == b.re && a.im == b.im;
+  inline operator ==(a: borrowed object?, b: borrowed object?) return __primitive("ptr_eq", a, b);
```

Comment:
I find myself wondering whether this case (object == object) will need to be made into an operator on the method in order for user-defined classes to define their own `==` overloads and have them work correctly w.r.t. this one in cases like https://github.com/Cray/chapel-private/issues/1731 (which remains my leading concern w.r.t. operator stability due to its subtlety).  That said, I think I also am conflating dynamic dispatch into that gut reaction incorrectly, and am wondering whether the same is true for other more value-based cases that might support user-defined overrides, like `enum == enum`.